CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/ui/DisplayNoteActivity.java

```diff
@@ -69,4 +69,14 @@ public void onSelectSaveNote(View v) {
         Intent intent = new Intent(this, MainActivity.class);
         startActivity(intent);
     }
+
+    public void onSelectDeleteNote(View v) {
+        Intent intent = getIntent();
+        String noteId = intent.getStringExtra(MainActivity.NOTE_ID);
+
+        noteService.delete(noteId);
+
+        intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
```

Comment:
Instead of starting new MainActivity you should call finish() to finish a current activity.